library/Freckle/App/Random.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Freckle.App.Random
+  ( smallRandomSubsetOfLargeIntegerRange
+  , Range (..)
+  , inclusiveRange
+  ) where
+
+import Freckle.App.Prelude
+
+import Control.Monad.Random (MonadRandom (..), Random)
+import Control.Monad.ST (runST)
+import Control.Monad.State.Strict (execStateT, get, put)
+import Data.Functor ((<&>))
+import Data.STRef (newSTRef, readSTRef, writeSTRef)
+import Numeric.Natural (Natural)
+
+import qualified Data.Set as Set
+
+-- | A contiguous range of integers
+data Range i = Range
+  { inclusiveMinBound :: i
+  , size :: Natural
```

Comment:
Would it be a nicer model to make this "offset to end" so that empty ranges are impossible?

So `Range N 0` would mean `[N, N+0]`

Reply:
That sounds nicer, yes.

Reply:
I think it's good to still have an empty range possibility, but I've reworked the type definitions such that an empty range no longer has a meaningless `inclusiveMinBound` number.